plugins/destination.go

```diff
@@ -162,14 +162,16 @@ func (p *DestinationPlugin) Write(ctx context.Context, tables schema.Tables, sou
 		select {
 		case <-gctx.Done():
 			close(ch)
-			return eg.Wait()
+			if err := eg.Wait(); err != nil {
+				return fmt.Errorf("context done %v. write client returned: %w", gctx.Err(), err)
+			}
 		case ch <- clientResource:
 		}
 	}
 
 	close(ch)
 	if err := eg.Wait(); err != nil {
-		return err
+		return fmt.Errorf("write client returned: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("write client failed with error: %w", err)
```

Also shouldn't we use the same message above, i.e. instead `plugin returned` -> `write failed`